### PR TITLE
[GLIMMER2] Migrate {{render}} tests

### PR DESCRIPTION
**Pull request by chadhietala**
This migrates the `{{render}}` tests to the new test infra. It also removes these tests dependency on the hardcoded `InteractiveRenderer` [in CoreView](https://github.com/emberjs/ember.js/blob/master/packages/ember-views/lib/views/core_view.js#L48-L52).

Ported, consolidated, and removed tests are annotated below.

Part of #13644

### CODE REVIEW

File: packages/ember-htmlbars/tests/helpers/render_test.js [hunk 1]
```diff
@@ -1,603 +0,0 @@
...
-  });
-
-  setTemplate('home', compile('<p>BYE</p>'));
-
-  runAppend(view);
-
-  equal(view.$().text(), 'HIBYE');
-  // This is a poor assertion. What is really being tested is that
-  // a second render with the same name will throw an assert.
-  ok(appInstance.lookup('router:main')._lookupActiveComponentNode('home'), 'should register home as active view');
```
Comment (chadhietala) [inline packages/ember-htmlbars/tests/helpers/render_test.js]: This just seems wrong so I removed it.

File: packages/ember-htmlbars/tests/helpers/render_test.js [hunk 2]
```diff
@@ -1,603 +0,0 @@
...
-});
-
-QUnit.test('{{render}} helper should render nested helpers', function() {
```
Comment (chadhietala) [inline packages/ember-htmlbars/tests/helpers/render_test.js]: [Ported](https://github.com/emberjs/ember.js/pull/13681/files#diff-6458c541b6838ccdc1da2f5a63c1c4a5R18)

File: packages/ember-htmlbars/tests/helpers/render_test.js [hunk 3]
```diff
@@ -1,603 +0,0 @@
...
-  setTemplate('foo', compile('<p>FOO</p>{{render \'bar\'}}'));
-  setTemplate('bar', compile('<p>BAR</p>{{render \'baz\'}}'));
-  setTemplate('baz', compile('<p>BAZ</p>'));
-
-  runAppend(view);
-
-  equal(view.$().text(), 'HIFOOBARBAZ');
-});
-
-QUnit.test('{{render}} helper should have assertion if neither template nor view exists', function() {
```
Comment (chadhietala) [inline packages/ember-htmlbars/tests/helpers/render_test.js]: [Ported](https://github.com/emberjs/ember.js/pull/13681/files#diff-6458c541b6838ccdc1da2f5a63c1c4a5R33)

File: packages/ember-htmlbars/tests/helpers/render_test.js [hunk 4]
```diff
@@ -1,603 +0,0 @@
...
-    controller: controller.create(),
-    template: compile(template)
-  });
-
-  expectAssertion(() => {
-    runAppend(view);
-  }, 'You used `{{render \'oops\'}}`, but \'oops\' can not be found as a template.');
-});
-
-QUnit.test('{{render}} helper should render given template with a supplied model', function() {
```
Comment (chadhietala) [inline packages/ember-htmlbars/tests/helpers/render_test.js]: [Ported](https://github.com/emberjs/ember.js/pull/13681/files#diff-6458c541b6838ccdc1da2f5a63c1c4a5R41)

File: packages/ember-htmlbars/tests/helpers/render_test.js [hunk 5]
```diff
@@ -1,603 +0,0 @@
...
-  equal(component.$().text(), 'HIRails is omakase');
-  equal(postController.get('model'), post);
-
-  runSet(component, 'post', { title: 'Rails is unagi' });
-
-  equal(component.$().text(), 'HIRails is unagi');
-  deepEqual(postController.get('model'), { title: 'Rails is unagi' });
-});
-
-QUnit.test('{{render}} helper with a supplied model should not fire observers on the controller', function () {
```
Comment (chadhietala) [inline packages/ember-htmlbars/tests/helpers/render_test.js]: [Ported](https://github.com/emberjs/ember.js/pull/13681/files#diff-6458c541b6838ccdc1da2f5a63c1c4a5R68)

File: packages/ember-htmlbars/tests/helpers/render_test.js [hunk 6]
```diff
@@ -1,603 +0,0 @@
...
-  appInstance.register('controller:post', PostController);
-
-  setTemplate('post', compile('<p>{{title}}</p>'));
-
-  let modelDidChange = 0;
-  runAppend(view);
-  equal(modelDidChange, 0, 'model observer did not fire');
-});
-
-QUnit.test('{{render}} helper should raise an error when a given controller name does not resolve to a controller', function() {
```
Comment (chadhietala) [inline packages/ember-htmlbars/tests/helpers/render_test.js]: [Ported](https://github.com/emberjs/ember.js/pull/13681/files#diff-6458c541b6838ccdc1da2f5a63c1c4a5R91)

File: packages/ember-htmlbars/tests/helpers/render_test.js [hunk 7]
```diff
@@ -1,603 +0,0 @@
...
-  });
-
-  setTemplate('home', compile('<p>BYE</p>'));
-
-  expectAssertion(() => {
-    runAppend(view);
-  }, 'The controller name you supplied \'postss\' did not resolve to a controller.');
-});
-
-QUnit.test('{{render}} helper should render with given controller', function() {
```
Comment (chadhietala) [inline packages/ember-htmlbars/tests/helpers/render_test.js]: [Ported](https://github.com/emberjs/ember.js/pull/13681/files#diff-6458c541b6838ccdc1da2f5a63c1c4a5R99)

File: packages/ember-htmlbars/tests/helpers/render_test.js [hunk 8]
```diff
@@ -1,603 +0,0 @@
...
-
-  let renderedController = appInstance.lookup('controller:posts');
-  let uniqueId = renderedController.get('uniqueId');
-  let renderedModel = renderedController.get('model');
-  equal(uniqueId, 0, 'precond - first uniqueId is used for singleton');
-  equal(uniqueId, view.$().html(), 'rendered with singleton controller');
-  equal(renderedModel, model, 'rendered with model on controller');
-});
-
-QUnit.test('{{render}} helper should rerender with given controller', function() {
```
Comment (chadhietala) [inline packages/ember-htmlbars/tests/helpers/render_test.js]: [Consolidated](https://github.com/emberjs/ember.js/pull/13681/files#diff-6458c541b6838ccdc1da2f5a63c1c4a5R99)

File: packages/ember-htmlbars/tests/helpers/render_test.js [hunk 9]
```diff
@@ -1,603 +0,0 @@
...
-  let renderedController = appInstance.lookup('controller:posts');
-  let uniqueId = renderedController.get('uniqueId');
-  let renderedModel = renderedController.get('model');
-
-  equal(uniqueId, 0, 'precond - first uniqueId is used for singleton');
-  equal(uniqueId, view.$().html(), 'rendered with singleton controller');
-  equal(renderedModel, model, 'rendered with model on controller');
-});
-
-QUnit.test('{{render}} helper should render a template without a model only once', function() {
```
Comment (chadhietala) [inline packages/ember-htmlbars/tests/helpers/render_test.js]: [Ported](https://github.com/emberjs/ember.js/pull/13681/files#diff-6458c541b6838ccdc1da2f5a63c1c4a5R237)

File: packages/ember-htmlbars/tests/helpers/render_test.js [hunk 10]
```diff
@@ -1,603 +0,0 @@
...
-  });
-
-  setTemplate('home', compile('<p>BYE</p>'));
-
-  expectAssertion(() => {
-    runAppend(view);
-  }, /\{\{render\}\} helper once/i);
-});
-
-QUnit.test('{{render}} helper should render templates with models multiple times', function() {
```
Comment (chadhietala) [inline packages/ember-htmlbars/tests/helpers/render_test.js]: [Ported](https://github.com/emberjs/ember.js/pull/13681/files#diff-6458c541b6838ccdc1da2f5a63c1c4a5R129)

File: packages/ember-htmlbars/tests/helpers/render_test.js [hunk 11]
```diff
@@ -1,603 +0,0 @@
...
-  equal(postController1.get('model'), post1);
-  equal(postController2.get('model'), post2);
-
-  runSet(component, 'post1', { title: 'I am new' });
-
-  ok(component.$().text().match(/^HI ?I am new ?Then me$/));
-  deepEqual(postController1.get('model'), { title: 'I am new' });
-});
-
-QUnit.test('{{render}} helper should not leak controllers', function() {
```
Comment (chadhietala) [inline packages/ember-htmlbars/tests/helpers/render_test.js]: Removed is this testing that the `View` doesn't leak the controller?

File: packages/ember-htmlbars/tests/helpers/render_test.js [hunk 12]
```diff
@@ -1,603 +0,0 @@
...
-  setTemplate('post', compile('<p>{{title}}</p>'));
-
-  runAppend(view);
-
-  runDestroy(view);
-
-  ok(postController.isDestroyed, 'expected postController to be destroyed');
-});
-
-QUnit.test('{{render}} helper should not treat invocations with falsy contexts as context-less', function() {
```
Comment (chadhietala) [inline packages/ember-htmlbars/tests/helpers/render_test.js]: [Ported](https://github.com/emberjs/ember.js/pull/13681/files#diff-6458c541b6838ccdc1da2f5a63c1c4a5R159)

File: packages/ember-htmlbars/tests/helpers/render_test.js [hunk 13]
```diff
@@ -1,603 +0,0 @@
...
-  setTemplate('post', compile('<p>{{#unless model}}NOTHING{{/unless}}</p>'));
-
-  runAppend(component);
-
-  ok(component.$().text().match(/^HI ?NOTHING ?NOTHING$/));
-  equal(postController1.get('model'), 0);
-  equal(postController2.get('model'), undefined);
-});
-
-QUnit.test('{{render}} helper should render templates both with and without models', function() {
```
Comment (chadhietala) [inline packages/ember-htmlbars/tests/helpers/render_test.js]: [Ported](https://github.com/emberjs/ember.js/pull/13681/files#diff-6458c541b6838ccdc1da2f5a63c1c4a5R174)

File: packages/ember-htmlbars/tests/helpers/render_test.js [hunk 14]
```diff
@@ -1,603 +0,0 @@
...
-  equal(postController1.get('model'), null);
-  equal(postController2.get('model'), post);
-
-  runSet(component, 'post', { title: 'Rails is unagi' });
-
-  ok(component.$().text().match(/^HI ?Title: ?Title:Rails is unagi$/));
-  deepEqual(postController2.get('model'), { title: 'Rails is unagi' });
-});
-
-QUnit.test('{{render}} helper should be able to render a template again when it was removed', function() {
```
Comment (chadhietala) [inline packages/ember-htmlbars/tests/helpers/render_test.js]: Removed. This seems very weird, but maybe this just needs to be an `ApplicationTest`?

File: packages/ember-htmlbars/tests/helpers/render_test.js [hunk 15]
```diff
@@ -1,603 +0,0 @@
...
-        template: compile('<div>2{{render \'home\'}}</div>')
-      }
-    };
-    view.setOutletState(liveRoutes);
-  });
-
-  equal(view.$().text(), 'HI2BYE');
-});
-
-QUnit.test('{{render}} works with dot notation', function() {
```
Comment (chadhietala) [inline packages/ember-htmlbars/tests/helpers/render_test.js]: [Ported](https://github.com/emberjs/ember.js/pull/13681/files#diff-6458c541b6838ccdc1da2f5a63c1c4a5R202)

File: packages/ember-htmlbars/tests/helpers/render_test.js [hunk 16]
```diff
@@ -1,603 +0,0 @@
...
-
-  setTemplate('blog.post', compile('{{uniqueId}}'));
-
-  runAppend(view);
-
-  let singletonController = appInstance.lookup('controller:blog.post');
-  equal(singletonController.uniqueId, view.$().html(), 'rendered with correct singleton controller');
-});
-
-QUnit.test('throws an assertion if {{render}} is called with an unquoted template name', function() {
```
Comment (chadhietala) [inline packages/ember-htmlbars/tests/helpers/render_test.js]: [Ported](https://github.com/emberjs/ember.js/pull/13681/files#diff-6458c541b6838ccdc1da2f5a63c1c4a5R218)

File: packages/ember-htmlbars/tests/helpers/render_test.js [hunk 17]
```diff
@@ -1,603 +0,0 @@
...
-  });
-
-  setTemplate('home', compile('<p>BYE</p>'));
-
-  expectAssertion(() => {
-    runAppend(view);
-  }, 'The first argument of {{render}} must be quoted, e.g. {{render "sidebar"}}.');
-});
-
-QUnit.test('throws an assertion if {{render}} is called with a literal for a model', function() {
```
Comment (chadhietala) [inline packages/ember-htmlbars/tests/helpers/render_test.js]: [Ported](https://github.com/emberjs/ember.js/pull/13681/files#diff-6458c541b6838ccdc1da2f5a63c1c4a5R226)

File: packages/ember-htmlbars/tests/helpers/render_test.js [hunk 18]
```diff
@@ -1,603 +0,0 @@
...
-  });
-
-  setTemplate('home', compile('<p>BYE</p>'));
-
-  expectAssertion(() => {
-    runAppend(view);
-  }, 'The second argument of {{render}} must be a path, e.g. {{render "post" post}}.');
-});
-
-QUnit.test('{{render}} helper should not require view to provide its own template', function() {
```
Comment (chadhietala) [inline packages/ember-htmlbars/tests/helpers/render_test.js]: Removed. `View` specific.

File: packages/ember-htmlbars/tests/helpers/render_test.js [hunk 19]
```diff
@@ -1,603 +0,0 @@
...
-  appInstance.register('template:fish', compile('Hello fish!'));
-
-  appInstance.register('view:fish', EmberView.extend());
-
-  runAppend(view);
-
-  equal(view.$().text(), 'Hello fish!');
-});
-
-QUnit.test('{{render}} helper should set router as target when parentController is not found', function() {
```
Comment (chadhietala) [inline packages/ember-htmlbars/tests/helpers/render_test.js]: [Ported](https://github.com/emberjs/ember.js/pull/13681/files#diff-6458c541b6838ccdc1da2f5a63c1c4a5R247)

File: packages/ember-glimmer/tests/integration/helpers/render-test.js [hunk 20]
```diff
@@ -0,0 +1,272 @@
...
+    this.registerTemplate('home', '<p>BYE</p>');
+    this.registerTemplate('foo', `<p>FOO</p>{{render 'bar'}}`);
+    this.registerTemplate('bar', `<p>BAR</p>{{render 'baz'}}`);
+    this.registerTemplate('baz', `<p>BAZ</p>`);
+
+    this.render('<h1>HI</h1>{{render \'foo\'}}');
+    this.assertText('HIFOOBARBAZ');
+  }
+
+  ['@test should have assertion if neither template nor view exists']() {
```
Comment (rwjblue) [inline packages/ember-glimmer/tests/integration/helpers/render-test.js]: Can you tweak this title and remove `view` (we don't support the view part anymore and I think I missed it when I updated te original test)?

**Pull request merged**